### ISSUE TITLE
Consider updating default version to latest release

### ISSUE THREAD

**Issue by tjlaboss** [idaholab/MontePy#462]
**Is your feature request related to a problem? Please describe.**
The default version of MCNP for MontePy is 6.2.0. At each new release supported by MontePy, consider updating the default version.

**Describe the solution you'd like**
As of issue submission, MCNP version 6.3.0 is the latest. In the documentation, prefer links to the 6.3.0 user manual. Update the global constant `montepy.DEFAULT_VERSION` to the tuple `(6,3,0)`.

**Describe alternatives you've considered**
 - Leave as 6.2.0 forever
 - Point to the earliest supported version instead

**Comment by MicahGale** [comment]
Good idea. I was holding off on this for #NaN, but I think we have had enough time to see this isn't a significant issue with 6.3.0. Though 6.3.1 I believe maybe.

**Comment by tjlaboss** [comment]
This could be automated...or Claudomated.

**Comment by MicahGale** [comment]
Yeah... I still don't like that we don't support 6.3.1 materials yet though.

**Comment by tjlaboss** [comment]
In my opinion:

The MCNP developers deprecate old features and syntax more rarely than adding new features and syntax. If previously valid syntax is removed, it is probably because it is error-prone or no longer useful; and a user who needs a specific deprecated feature is probably aware of the version limitation for some niche workflow. Furthermore, I suspect that users locked into older versions are less likely to use MontePy than those running recent versions.

Therefore, MontePy's default version should be the latest release _that is implemented_. i.e., once MCNP 6.3.1 materials are supported, the default version should be revised to `(6, 3, 1)`.
